cereus/src/main/java/com/cactus/adapters/AuthLogin.java

```diff
@@ -0,0 +1,186 @@
+     * The function also takes in a Universal Resource Identifier (URI), the location of where the HttpRequest is sent,
+     * and the string and URI are passed into a HttpRequest.
+     *
+     * After the request is sent, the response is stored in a HttpResponse object. Then the body of the
+     * response is mapped into a User object, which is returned
+     *
+     * @param body     a Hashmap containing the required properties for the HttpRequest
+     * @param uri      a URI object that gives the location of the request
+     * @return         a User object created from the response object
+     *
+     * @see User
+     */
+    private User sendRequest(HashMap<String, String> body, URI uri) throws IOException, InterruptedException {
+        ObjectMapper objectMapper = new ObjectMapper();
+        String requestBody = objectMapper.writeValueAsString(body);
+        HttpClient client = HttpClient.newHttpClient();
+        HttpRequest request = HttpRequest.newBuilder()
+                .uri(uri)
+                .header("Content-Type", "application/json")
+                .POST(HttpRequest.BodyPublishers.ofString(requestBody))
+                .build();
+        HttpResponse<String> response = client.send(request, HttpResponse.BodyHandlers.ofString());
+        ObjectMapper finalMapper =
+                new ObjectMapper().configure(DeserializationFeature.FAIL_ON_UNKNOWN_PROPERTIES, false);
+        return finalMapper.readValue(response.body(), User.class);
+    }
+
+
+    /**
+     * Returns whether the User object with the corresponding token is successfully logged out of the application.
+     *
+     * The token of a User object is sent to a server, which logs out the User corresponding ot the token.
+     * If the token corresponds to no User object, then the User is not logged out and the function returns false.
+     *
+     * @param token    a String containing a token unique to every User, stored in the User class
+     * @return         a boolean signifying whether the User corresponding to the token is logged out or not
+     *
+     * @see User
+     */
+    @Override
+    public boolean logout(String token) {
+        HttpClient client = HttpClient.newHttpClient();
+        URI uri;
+        try{
+            uri = new URI("http://localhost:8080/logout");
+        }
+        catch(URISyntaxException u){
+            return false;
+        }
+        HttpRequest request = HttpRequest.newBuilder()
+                .uri(uri)
+                .setHeader("Authorization", token)
+                .header("Content-Type", "application/json")
+                .POST(HttpRequest.BodyPublishers.ofString(""))
+                .build();
+        HttpResponse<String> response;
+        try {
+            response = client.send(request, HttpResponse.BodyHandlers.ofString());
+        }
+        catch(IOException | InterruptedException i){
+            return false;
+        }
+        return response.statusCode() == 204;
```

Comment:
Consider using an enum here for readability (instead of just `204`), I'm sure one exists for HTTP status codes in either base Java or the Apache library.